raiden/smart_contracts/TokenNetworksRegistry.sol

```diff
@@ -0,0 +1,34 @@
+pragma solidity ^0.4.17;
+
+import "./Token.sol";
+import "./TokenNetwork.sol";
+import "./Utils.sol";
+
+contract TokenNetworkRegistry is Utils {
```

Comment:
Right now we have the following functions for the Registry, do you plan to add them?
- `channelManagerByToken(address token_address)`, would become `getTokenNetworkForToken(...)`
- `channelManagerAddresses()`, would become `getTokenNetworkAddresses()`

Reply:
In general, the client will get data from events to save gas. At least this was agreed upon during the spec discussions.
- `channelManagerByToken` can now be replaced with `TokenNetworkRegistry.token_to_token_networks(token_address )`
- `channelManagerAddresses` - you get that from the `TokenNetworkCreated` events

Reply:
Ah, I missed that `token_to_token_networks` was public. For getting events efficiently, would it make sense to store the contract creation block number somewhere?

Reply:
I don't think this should be done in the SC. You can store/hardcode the block number at which the `TokenNetworkRegistry` was deployed in the Py client and use that for the `TokenNetworkCreated` events -> get the block number for each deployed `TokenNetwork`

Reply:
WE need to discuss the "getting data from events" idea.

As we can see in practise it does not work for one of the major clients. Parity. Parity can only get events from the block it gets ouf of warp and after. If we say we don't care about that -- then we don't support parity unless it does not use warp.

That's assuming they have not changed this behaviour in latest parity versions.

Reply:
I'm am a bit skeptical with this idea TBH